pkg/querier/frontend/results_cache.go

```diff
@@ -110,7 +110,7 @@ func (s resultsCache) handleHit(ctx context.Context, r *QueryRangeRequest, exten
 		return response, nil, err
 	}
 
-	reqResps, err = doRequests(ctx, s.next, requests)
+	reqResps, err = doRequests(ctx, s.next, requests, len(requests))
```

Comment:
Why is this `len(requests)` instead of `maxParallelism`? If it can be `maxParallelism` then I think the function signature will not even need to change.

Reply:
In this case I thought it would be easier to ensure the requests would not have to wait, but I guess we can simplify.

Reply:
:+1: